### DISCUSSION TITLE
Why reset_parameters?

### DISCUSSION THREAD

**Discussion by Jeriousman** [pyg-team/pytorch_geometric#4006] [Q&A] [answered]
I see reset_parameters() in nn layers. It seems that it is 'a must' to do in GNN or PyG. Why should we use it?

**Comment by rusty1s** [comment, marked as answer]
It isn't a must. It's just a convenient method to reset the parameters of a model, e.g., for re-using the model across different model initializations. It also follows best-practice from PyTorch, see [here](https://pytorch.org/docs/stable/_modules/torch/nn/modules/linear.html#Linear).

**Reply by Jeriousman** [reply]
Thank you. I will check the link!
